cmd/api.go

```diff
@@ -163,41 +163,15 @@ func (o *apiOptions) validate(c *config.Config) error {
 
 	// In case we want to scaffold a resource API we need to do some checks
 	if o.doResource {
-		// Skip the following check for v1 as resources aren't tracked
-		if !c.IsV1() {
-			// Check that resource doesn't exist or flag force was set
-			if !o.force {
-				resourceExists := false
-				for _, r := range c.Resources {
-					if r.Group == o.resource.Group &&
-						r.Version == o.resource.Version &&
-						r.Kind == o.resource.Kind {
-						resourceExists = true
-						break
-					}
-				}
-				if resourceExists {
-					return errors.New("API resource already exists")
-				}
-			}
+		// Check that resource doesn't exist or flag force was set
+		if !o.force && c.HasResource(o.resource.GVK()) {
+			return errors.New("API resource already exists")
 		}
 
-		// The following check is v2 specific as multi-group isn't enabled by default
-		if c.IsV2() {
-			// Check the group is the same for single-group projects
-			if !c.MultiGroup {
-				validGroup := true
-				for _, existingGroup := range c.ResourceGroups() {
-					if !strings.EqualFold(o.resource.Group, existingGroup) {
-						validGroup = false
-						break
-					}
-				}
-				if !validGroup {
-					return fmt.Errorf("multiple groups are not allowed by default, to enable multi-group visit %s",
-						"kubebuilder.io/migration/multi-group.html")
-				}
-			}
+		// Check that the provided group can be added to the project
+		if !c.IsGroupAllowed(o.resource.Group) {
+			return fmt.Errorf("multiple groups are not allowed by default, to enable multi-group visit %s",
```

Comment:
Since it is a func now, thee specific error should came from the `IsGroupAllowed` because it could have more than one check/error.

Reply:
`IsGroupAllowed` doesn't have any error. You will be able to call that function anytime and it won't break. The error is in validating the api that doesn't fulfill one of the requirements (the group being allowed).

Reply:
Since`IsGroupAllowed`  can do more than one check, so why it would have here a 1  specific error?

```
if !c.IsGroupAllowed(o.resource.Group) {
			return fmt.Errorf("multiple groups are not allowed by default, to enable multi-group visit %s",
 ...
```

Reply:
There is only one reason why the group may not be allowed, being v2, single-group, and already existing a group with a different name. The other checks return `true`, they are just shot-circuits that we need to have to support different versions and multi-group v2.

Reply:
Currently, yes. However, the change allows it to be introduced. Another option is to make clear what will be checked by replacing the name for `isMultiGroupAllowed`. I would prefer this one.

Reply:
Summing up the offline discussion: having a generic method `IsGroupAllowed` being the only check for that error will mean that if in the future another check is added to that function, an unexpected error may be returned. However moving the api-specific error to the `model` package doesn't seem right either. Instead we moved to a `HasGroup` approach with extra checks in the API side to check for v2 and single-group.